### PR TITLE
bpf: complexity-tests: add HAVE_FIB_NEIGH 

### PR DESCRIPTION
**Pull request by julianwiedmann**
Reflect HAVE_FIB_NEIGH in the relevant kernels. While at it fix up a few left-overs.

### CODE REVIEW

**Comment by julianwiedmann** [comment]
@dylandreimerink could you keep me honest here? :pray: 

For reference, I believe we're still missing these features:

> pkg/datapath/linux/probes/probes.go:592:                "HAVE_SET_RETVAL":       probes.ProgramHelpers[ProgramHelper{ebpf.CGroupSock, asm.FnSetRetval}]
> pkg/datapath/linux/probes/probes.go:603:                "HAVE_CHANGE_TAIL": probes.ProgramHelpers[ProgramHelper{ebpf.SchedCLS, asm.FnSkbChangeTail}]
> pkg/datapath/linux/probes/probes.go:605:                "HAVE_CSUM_LEVEL":  probes.ProgramHelpers[ProgramHelper{ebpf.SchedCLS, asm.FnCsumLevel}]

**Review by borkmann — APPROVED** [review]

**Review by aspsk — APPROVED** [review]

**Comment by dylandreimerink** [comment]
> @dylandreimerink could you keep me honest here? 🙏

As far as I can tell this is all correct, both flags are expected for everything >=5.10

> For reference, I believe we're still missing these features:

>>pkg/datapath/linux/probes/probes.go:592: "HAVE_SET_RETVAL": probes.ProgramHelpers[ProgramHelper{ebpf.CGroupSock, asm.FnSetRetval}]
pkg/datapath/linux/probes/probes.go:603: "HAVE_CHANGE_TAIL": probes.ProgramHelpers[ProgramHelper{ebpf.SchedCLS, asm.FnSkbChangeTail}]
pkg/datapath/linux/probes/probes.go:605: "HAVE_CSUM_LEVEL": probes.ProgramHelpers[ProgramHelper{ebpf.SchedCLS, asm.FnCsumLevel}]

We are also missing `HAVE_CGROUP_ID`, `HAVE_SOCKET_LOOKUP`, `HAVE_JIFFIES`, `HAVE_SOCKET_COOKIE`, and `HAVE_NETNS_COOKIE` in the complexity check, not sure if these have significant impact though. 

And the `HAVE_FIB_LOOKUP` macro isn't used in the bpf code if my grep is correct. So we should be able to remove the probe code for that as well.

**Review by rgo3 — APPROVED** [review]
Do we want to remove the respective `HAVE_FIB_LOOKUP` probe code with this PR or a follow-up? If follow-up, I can put it on my list :)

**Comment by julianwiedmann** [comment]
> Do we want to remove the respective `HAVE_FIB_LOOKUP` probe code with this PR or a follow-up? If follow-up, I can put it on my list :)

let's do it in a follow-up, not sure that's something we would want to be part of the backport.

File: bpf/Makefile [hunk 1]
```diff
@@ -85,12 +85,12 @@ HOST_OPTIONS = $(LXC_OPTIONS) \
 	-DDISABLE_LOOPBACK_LB:-DENABLE_IPV4:-DENABLE_IPV6:-DENCAP_IFINDEX:-DTUNNEL_MODE:-DPOLICY_VERDICT_NOTIFY:-DENABLE_HOST_FIREWALL: \
 	-DDISABLE_LOOPBACK_LB:-DENABLE_IPV4:-DENABLE_IPV6:-DENCAP_IFINDEX:-DTUNNEL_MODE:-DPOLICY_VERDICT_NOTIFY:-DENABLE_DSR: \
 	-DDISABLE_LOOPBACK_LB:-DENABLE_IPV4:-DENABLE_IPV6:-DENCAP_IFINDEX:-DTUNNEL_MODE:-DPOLICY_VERDICT_NOTIFY:-DENABLE_IPSEC:-DENABLE_DSR: \
-	-DDISABLE_LOOPBACK_LB:-DENABLE_IPV4:-DENABLE_IPV6:-DENCAP_IFINDEX:-DTUNNEL_MODE:-DPOLICY_VERDICT_NOTIFY:-DENABLE_IPSEC:-DHAVE_FIB_LOOKUP:-DENABLE_DSR: \
+	-DDISABLE_LOOPBACK_LB:-DENABLE_IPV4:-DENABLE_IPV6:-DENCAP_IFINDEX:-DTUNNEL_MODE:-DPOLICY_VERDICT_NOTIFY:-DENABLE_IPSEC:-DENABLE_DSR: \
```
Comment (tklauser) [inline bpf/Makefile:88]: This option combination is now the same as the one on the previous line, so one of them can be dropped I think.

File: bpf/Makefile [hunk 2]
```diff
@@ -85,12 +85,12 @@ HOST_OPTIONS = $(LXC_OPTIONS) \
...
 	-DDISABLE_LOOPBACK_LB:-DENABLE_IPV4:-DENABLE_IPV6:-DENCAP_IFINDEX:-DTUNNEL_MODE:-DPOLICY_VERDICT_NOTIFY:-DENABLE_NODEPORT:-DENABLE_MASQUERADE_IPV4:-DENABLE_MASQUERADE_IPV6: \
 	-DDISABLE_LOOPBACK_LB:-DENABLE_IPV4:-DENABLE_IPV6:-DENCAP_IFINDEX:-DTUNNEL_MODE:-DPOLICY_VERDICT_NOTIFY:-DENABLE_NODEPORT:-DENABLE_MASQUERADE_IPV4:-DENABLE_MASQUERADE_IPV6:-DENABLE_EGRESS_GATEWAY: \
 	-DDISABLE_LOOPBACK_LB:-DENABLE_IPV4:-DENABLE_IPV6:-DENCAP_IFINDEX:-DTUNNEL_MODE:-DPOLICY_VERDICT_NOTIFY:-DENABLE_NODEPORT:-DENABLE_MASQUERADE_IPV4:-DENABLE_MASQUERADE_IPV6:-DENABLE_EGRESS_GATEWAY_COMMON: \
 	-DDISABLE_LOOPBACK_LB:-DENABLE_IPV4:-DENABLE_IPV6:-DENCAP_IFINDEX:-DTUNNEL_MODE:-DPOLICY_VERDICT_NOTIFY:-DENABLE_IPSEC:-DENABLE_NODEPORT:-DENABLE_MASQUERADE_IPV4:-DENABLE_MASQUERADE_IPV6: \
-	-DDISABLE_LOOPBACK_LB:-DENABLE_IPV4:-DENABLE_IPV6:-DENCAP_IFINDEX:-DTUNNEL_MODE:-DPOLICY_VERDICT_NOTIFY:-DENABLE_IPSEC:-DHAVE_FIB_LOOKUP:-DENABLE_NODEPORT:-DENABLE_MASQUERADE_IPV4:-DENABLE_MASQUERADE_IPV6: \
+	-DDISABLE_LOOPBACK_LB:-DENABLE_IPV4:-DENABLE_IPV6:-DENCAP_IFINDEX:-DTUNNEL_MODE:-DPOLICY_VERDICT_NOTIFY:-DENABLE_IPSEC:-DENABLE_NODEPORT:-DENABLE_MASQUERADE_IPV4:-DENABLE_MASQUERADE_IPV6: \
```
Comment (tklauser) [inline bpf/Makefile:93]: Same here, this is now the same as the previous line.

File: bpf/Makefile (hunk 1 above)
Comment (julianwiedmann) [reply bpf/Makefile:88]: :see_no_evil: thanks for spotting, I'll go clean that up.

**Pull request merged**